datafusion/sql/src/planner.rs

```diff
@@ -86,6 +88,35 @@ impl IdentNormalizer {
     }
 }
 
+/// Value Normalizer
+#[derive(Debug)]
+pub struct ValueNormalizer {
+    normalize: bool,
+}
+
+impl Default for ValueNormalizer {
+    fn default() -> Self {
+        Self { normalize: true }
+    }
+}
+
+impl ValueNormalizer {
+    pub fn new(normalize: bool) -> Self {
+        Self { normalize }
+    }
+
+    pub fn normalize(&self, value: Value) -> Result<String> {
+        if self.normalize {
+            crate::utils::normalize_value(&value)
+        } else {
+            match crate::utils::value_to_string(&value) {
+                Some(s) => Ok(s),
+                None => internal_err!("Unsupport value type to string: {:?}", value),
+            }
+        }
+    }
+}
```

Comment:
```suggestion
    pub fn normalize(&self, value: String) -> String {
        if self.normalize {
            value.to_ascii_lowercase()
        } else {
            value
        }
    }
}
```

I don't think the normalize function should return an error. How about we do the string conversion and normalization in separate steps in the `parse_options_map` method. 

Something like:
```rs
let value_string = match value_to_string(&value) {
  Some(s) => self.value_normalizer.normalize(s),
  None => return plan_err!("Unsupported Value in Options : {}", value),
};
```

Reply:
Done.